CorsixTH/Src/th_lua_gfx.cpp

```diff
@@ -277,11 +280,23 @@ int l_freetype_font_new(lua_State* L) {
 }
 
 int l_freetype_font_set_spritesheet(lua_State* L) {
+  // Colour 0 falls back to using the average colour of the sprite sheet.
+  argb_colour colour = 0;
+
   freetype_font* pFont = luaT_testuserdata<freetype_font>(L);
   sprite_sheet* pSheet = luaT_testuserdata<sprite_sheet>(L, 2);
+  if (!lua_isnoneornil(L, 3)) {
+    int red = static_cast<int>(luaL_checkinteger(L, 3));
+    int green = static_cast<int>(luaL_checkinteger(L, 4));
+    int blue = static_cast<int>(luaL_checkinteger(L, 5));
+    red = std::max(0, std::min(255, red));
```

Comment:
good excuse for `int alpha = static_cast<int>(lua_optinteger(L, 6, 255));` ?
not sure anyone would ever want that... probably not.